chore: removing unnecessary lowercase conversion (fixes latex) + some nits

signorecello opened the pull request:
There was this leftover thing from my experiments serving the doxygen page on docusaurus. It was breaking LaTeX when served directly (as it is now).

This fixes it.

ludamad commented on barretenberg/docs/docs/index.md:
```diff
@@ -66,7 +65,6 @@ bb.js is a TypeScript library that wraps the Barretenberg C++ implementation via
-- Smaller proof sizes
 - Reduced memory usage
 - Enhanced recursive proving capabilities
 
```
`- Enhanced recursive proving capabilities` what does that mean?

signorecello replied:
As you can imagine this was AI generated. All I needed was a page to show at the top 😅

The whole content of the BB docs is pretty much a WIP

ludamad replied:
yeah no worries just trying to improve the content - but maybe just put lorem ipsum if the intent is not to be reviewed?

ludamad reviewed (approved):
lgtm just use label `ci-squash-and-merge` when ready

signorecello committed:
chore: removing unnecessary lowercase conversion (fixes latex) + some nits

There was this leftover thing from my experiments serving the doxygen page on docusaurus. It was breaking LaTeX when served directly (as it is now).

This fixes it.
